src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/blockeditor/blockeditor.controller.js

```diff
@@ -0,0 +1,78 @@
+//used for the media picker dialog
+angular.module("umbraco")
+.controller("Umbraco.Editors.BlockEditorController",
+    function ($scope, localizationService, formHelper) {
+        var vm = this;
+
+        vm.content = $scope.model.content;
+        vm.settings = $scope.model.settings;
+
+        localizationService.localizeMany([
+            $scope.model.liveEditing ? "prompt_discardChanges" : "general_close",
+            $scope.model.liveEditing ? "buttons_confirmActionConfirm" : "buttons_submitChanges"
+        ]).then(function (data) {
+            vm.closeLabel = data[0];
+            vm.submitLabel = data[1];
+        });
+
+        vm.model = $scope.model;
+
+        vm.tabs = [];
+
+        if (vm.content && vm.content.variants) {
+
+            var apps = vm.content.apps;
+
+            vm.tabs = apps;
+
+            // replace view of content app.
+            var contentApp = apps.find(entry => entry.alias === "umbContent");
+            if(contentApp) {
+                contentApp.view = "views/common/infiniteeditors/elementeditor/elementeditor.content.html";
+                if($scope.model.hideContent) {
+                    apps.splice(apps.indexOf(contentApp), 1);
+                } else if ($scope.model.openSettings !== true) {
+                    contentApp.active = true;
+                }
+            }
+
+            // remove info app:
+            var infoAppIndex = apps.findIndex(entry => entry.alias === "umbInfo");
+            apps.splice(infoAppIndex, 1);
+
+        }
+
+        if (vm.settings && vm.settings.variants) {
+            localizationService.localize("blockEditor_tabBlockSettings").then(
+                function (settingsName) {
+                    var settingsTab = {
+                        "name": settingsName,
+                        "alias": "settings",
+                        "icon": "icon-settings",
+                        "view": "views/common/infiniteeditors/elementeditor/elementeditor.settings.html"
```

Comment:
This is strange, why does this render a view from somewhere else and this is the only place where that view is used? Shouldn't that view be moved to the same folder as this controller?

Reply:
Yes, so I actually already corrected this when looking at the naming. So I think this is better now.